mara_pipelines/logging/run_log.py

```diff
@@ -99,10 +106,17 @@ def close_open_run_after_error(run_id: int):
 
 
 class RunLogger(events.EventHandler):
+    """A base class for a run logger of pipeline events"""
```

Comment:
Not sure if this is really the right split: e.g. the logic for collecting lines is in the dbrunlogger and not here. This feels more like a NullLogger or so and dbrunlogger should get the run_id... Also the system statistics is still part of this.

Reply:
I did a redesign on this now. Please check again.

When it comes to the system statistics: I think this should be a separate PR, see here: https://github.com/mara/mara-pipelines/issues/72